### PR TITLE
feat(alertview): add alert view and action sheet support in swiftUI

### PR DESCRIPTION
**Pull request by ghost**
## Description:

1.  Adding alert view and action sheet support in SwiftUI.

## Code Usage:

```swift
import SwiftUI
import FueledUtilsSwiftUI

struct ContentView: View {
	@State private var showAlertModal = false
	@State private var showActionSheet = false

	let alertModel = AlertModel(
		title: "Alert Title",
		message: "Alert Message",
		buttons: [
			AlertButton(title: "Okay", action: {
				print("Okay")
			}),
			AlertButton(title: "Cancel", action: {
				print("Cancel")
			})
		]
	)

	let actionSheetModel = AlertModel(
		title: "Action Sheet Title",
		message: "Action Sheet Message",
		buttons: [
			AlertButton(title: "Option 1", action: {
				print("Option 1")
			}),
			AlertButton(title: "Option 2", action: {
				print("Option 2")
			}),
			AlertButton(title: "Option 3", action: {
				print("Option 3")
			})
		]
	)

	var body: some View {
		VStack {
			Spacer()
			Image(systemName: "globe")
				.imageScale(.large)
				.foregroundColor(.accentColor)
			Text("Show Alert")
				.foregroundColor(.white)
				.padding()
				.background(Color.black)
				.onTapGesture {
					showAlertModal.toggle()
				}
			Text("Show Action Sheet")
				.foregroundColor(.white)
				.padding()
				.background(Color.black)
				.onTapGesture {
					showActionSheet.toggle()
				}
			HStack {
				Spacer()
			}
			Spacer()
		}
		.background(Color.white)
		.showAlert(alertModel: alertModel, isPresented: $showAlertModal)
		.showAlert(
			alertModel: actionSheetModel,
			presentationStyle: .bottomSheet(titleVisibility: .visible),
			isPresented: $showActionSheet
		)
	}
}
```

### CODE REVIEW

File: FueledUtils/SwiftUI/AlertView.swift [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+				actions: {
+					ForEach($0.buttons) { button in
+						Button(button.title, action: button.action)
+					}
+				},
+				message: {
+					if let message = $0.message {
+						Text(message)
+					}
+				}
```
Comment (harishsaini) [inline FueledUtils/SwiftUI/AlertView.swift:99]: we have two different ways of creating message param for alert and confirmationDialog, can we please stick to one? (PS: IMO this way is better)

File: FueledUtils/SwiftUI/AlertView.swift [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+			self.action = action
+		}
+	}
+}
+
+@available(iOS 15.0, *)
+extension AlertButton: Identifiable {
+	public var id: String {
+		title
+	}
```
Comment (harishsaini) [inline FueledUtils/SwiftUI/AlertView.swift:141]: same title could be passed by mistake, probably better to use
```
	let id = UUID()
```
what do you think?

Comment (ghost) [reply FueledUtils/SwiftUI/AlertView.swift:141]: Make sense

File: FueledUtils/SwiftUI/AlertView.swift (hunk 1 above)
Comment (ghost) [reply FueledUtils/SwiftUI/AlertView.swift:99]: Alert Text is required field but in confirmation it's optional

File: FueledUtils/SwiftUI/AlertView.swift (hunk 2 above)
Comment (ghost) [reply FueledUtils/SwiftUI/AlertView.swift:141]: @harishsaini : Can you update review to request change as it becomes protected and not able to push changes

Comment (harishsaini) [reply FueledUtils/SwiftUI/AlertView.swift:141]: what do you mean? not able to push changes?

Comment (harishsaini) [reply FueledUtils/SwiftUI/AlertView.swift:141]: I have not approved, only commented to suggest minor changes, what issue are you facing due to that? 🤔

Comment (ghost) [reply FueledUtils/SwiftUI/AlertView.swift:141]: Yep, not able to

Comment (harishsaini) [reply FueledUtils/SwiftUI/AlertView.swift:141]: @ahujamanish Do you know what is happening here?

Comment (harishsaini) [reply FueledUtils/SwiftUI/AlertView.swift:141]: Ok , requesting changes to see if it works

**Review by harishsaini — CHANGES_REQUESTED** [review]
please see comments

**Review by vinaykharb — APPROVED** [review]
Looks good to me!!

**Review by ahujamanish — APPROVED** [review]

**Pull request closed without merging**